src/utils/fundamentals/fundamentals.js

```diff
@@ -0,0 +1,199 @@
+import * as utils from '../network/utils'
+import { searchData } from '../network'
+
+/**
+ * Make request and search data
+ * @function countries
+ * @param {object} params - object of parameters:
+ * @param {string} params.countryCode 
+ * @param {function} [callback] - Called with two arguments (error, result) on completion. `result` is a country name returned or country names. Omit to return Promise
+ * @returns {Promise|null} 
+ */
+export function countries({ code }, callback) {
+  const params = {
+    AMaaSClass: 'fundamentalCountries',
+    query: {countryCode: [code]}
+  }
+  //if code is not specified, retun all countries 
+  let promise = searchData(params).then(result => {
+    if (typeof callback === 'function') {
+      callback(null, result)
+    }
+    return result
+   }).catch((err) => {
+     console.log(err.message);
+  });
+  if (typeof callback !== 'function') {
+    // return promise if callback is not provided
+    return promise
+  }
+  promise.catch(error => callback(error))
+}
+
+/**
+ * Make request and search data
+ * @function calcBusinessDate
+ * @param {object} params - object of parameters:
+ * @param {string} params.codes - A country code string or an array of country codes strings 
+ * @param {string} params.date - start date
+ * @param {string} params.offset
+ * @param {string} params.invalidDates - an invalid date string or an array of invalid dates strings 
+ * @param {function} [callback] - Called with two arguments (error, result) on completion. `result` is a country name returned or country names. Omit to return Promise
+ * @returns {Promise|null} 
+ */
+export function calcBusinessDate({date, codes, offset, invalidDates}, callback) //invalid dates
+{
+    if(!date){
+        return "Must specify start_date in querystring"
+    }else if(!codes){
+        return "Must specify at least one country_code in querystring"
+    }else if(!offset){
+        return "Must specify offset in querystring"
+    }else if(!Number.isInteger(offset)) {
+         return "Offset must be an integer"
+    }
+
+    let code=""
+    if(Array.isArray(codes)){
```

Comment:
If `codes` is an array, you do not need to do anything special. The `searchData` function already accepts an array. So if `codes` is an array, then in your params on line 87, you will just have:

```javascript
params = {
  query: { /* other data */ countryCode: codes, /* other data */
}
```

If `codes` is not an array, we can convert it easily using `split()`:

```javascript
if (!Array.isArray(codes)) {
  codes = codes.split(',')
// This will convert 'code1,code2,code3' into ['code1', 'code2', 'code3']
// If codes is just 'code1', then it will convert it to ['code1]
}
```